source/common/upstream/health_checker_base_impl.cc

```diff
@@ -257,6 +257,15 @@ void HealthCheckerImplBase::ActiveHealthCheckSession::handleSuccess(bool degrade
     }
   }
 
+  // Clear the pending flag if it is set. By removing this we're marking the host as having been
```

Comment:
nit: maybe put this logic into a helper function that can be shared here and below? This would keep the comments and reasoning in one place? WDYT?